servers/quarkus-server/src/main/java/org/projectnessie/server/authn/NessieHttpAuthenticator.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.server.authn;
+
+import io.quarkus.security.AuthenticationFailedException;
+import io.quarkus.security.identity.IdentityProvider;
+import io.quarkus.security.identity.IdentityProviderManager;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.identity.request.AnonymousAuthenticationRequest;
+import io.quarkus.vertx.http.runtime.security.HttpAuthenticationMechanism;
+import io.quarkus.vertx.http.runtime.security.HttpAuthenticator;
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.ext.web.RoutingContext;
+import javax.annotation.Priority;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Alternative;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+/**
+ * A custom {@link HttpAuthenticator}. This authenticator that performs the following main duties:
+ *
+ * <ul>
+ *   <li>Prevents the Quarkus OIDC authentication mechanism from attempting authentication when it
+ *       is not configured.
+ *   <li>Completely disallows unauthenticated requests when authentication is enabled.
+ * </ul>
+ */
+@Alternative // @Alternative + @Priority ensure the original HttpAuthenticator bean is not used
+@Priority(1)
+@ApplicationScoped
+public class NessieHttpAuthenticator extends HttpAuthenticator {
```

Comment:
I think, the mechanism should really live inside Quarkus. Wanna raise a PR against Quarkus to add a runtime-config-flag there? WDYT?

Reply:
Good point. Let me think about how we can structure that in Quarkus code :thinking:

Reply:
I hope the latest version of `NessieHttpAuthenticator` is cleaner. We do not have to work around any Quarkus-specific behaviour now. I guess we do not really need any quarkus-side fixes now. WDYT?

Reply:
I'm okay with that.